src/ngraph/builder/autobroadcast.cpp

```diff
@@ -150,39 +150,38 @@ namespace ngraph
         /// If necessary, wrap \p node with an additional reshape and/or broadcast op.
         /// Return a pointer to the node that produces the wrapped value.
         /// If no additional reshape or broadcast op was needed, simply return \p node.
-        static std::shared_ptr<Node>
-            add_required_ops(const std::shared_ptr<Node>& node,
-                             const ngraph::Shape& node_shape_after_possible_reshaping,
-                             const ngraph::AxisSet& node_broadcast_axes,
-                             const ngraph::Shape& node_final_shape)
+        static Output<Node> add_required_ops(const Output<Node>& value,
+                                             const ngraph::Shape& shape_after_possible_reshaping,
+                                             const ngraph::AxisSet& broadcast_axes,
+                                             const ngraph::Shape& final_shape)
         {
-            std::shared_ptr<Node> return_node{node};
+            Output<Node> return_value{value};
 
-            if (node->get_shape() != node_shape_after_possible_reshaping)
+            if (value.get_shape() != shape_after_possible_reshaping)
             {
                 // tell reshape to examine input dimensions in order
-                ngraph::AxisVector order = ngraph::get_default_order(node->get_shape());
-                return_node = std::make_shared<ngraph::op::Reshape>(
-                    return_node, order, node_shape_after_possible_reshaping);
+                ngraph::AxisVector order = ngraph::get_default_order(value.get_shape());
+                return_value = std::make_shared<ngraph::op::Reshape>(
+                    return_value, order, shape_after_possible_reshaping);
             }
 
-            if (node_final_shape != node_shape_after_possible_reshaping)
+            if (final_shape != shape_after_possible_reshaping)
             {
-                return_node = std::make_shared<ngraph::op::Broadcast>(
-                    return_node, node_final_shape, node_broadcast_axes);
+                return_value = std::make_shared<ngraph::op::Broadcast>(
+                    return_value, final_shape, broadcast_axes);
             }
 
-            return return_node;
+            return return_value;
         }
 
-        std::pair<std::shared_ptr<Node>, std::shared_ptr<Node>>
-            numpy_broadcast(const std::pair<std::shared_ptr<Node>, std::shared_ptr<Node>>& args)
+        std::pair<Output<Node>, Output<Node>>
+            numpy_broadcast_values(const std::pair<Output<Node>, Output<Node>>& args)
```

Comment:
Instead of adding these functions with the `_values` suffix, could you try keeping the existing ones and having them keep `Output<Node>` as arguments but return `shared_ptr<Node>` as results?

If that doesn't work: I'm a little worried that the `_values` suffix on the function names is hard to understand. Maybe `_from_outputs`?